### PR TITLE
Spark hdfs test

### PR DESCRIPTION
**Pull request by snocke**
https://ci.stackable.tech/view/99%20Dev%20'n%20Demo/job/Functional-Cluster-tests/73/console

### CODE REVIEW

File: kuttl-test/tests/.kubeassert/spark_file_exist.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ function spark_file_exist {
 
     # Validate results
     if [ "${SPARK_FILE}" == $1 ]; then
-      #echo "$SPARK_FILE exists."
+      #Print success message
```
Comment (maltesander) [inline kuttl-test/tests/.kubeassert/spark_file_exist.sh:23]: remove the comment?

File: kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh [hunk 2]
```diff
@@ -11,17 +11,20 @@ function spark_hdfs_file_exist {
     [[ -z $1 ]] && logger::error "You must specify a fully qualified file name '<file-name>.<type>'" && exit 1
     # Print assertion message
     logger::assert "File with name $1 should be on path."
+
     # Run some kubectl commands
+    # TODO: Find a better way then this to get the podname
+    # export NAME_NODE_POD=$(kubectl -n ${NAMESPACE} get pods -o=name | grep hdfs-namenode-default-0 | sed "s/^.\{4\}//")
```
Comment (maltesander) [inline kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh:17]: The TODO still relevant?

File: kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh [hunk 3]
```diff
@@ -11,17 +11,20 @@ function spark_hdfs_file_exist {
...
     export NAME_NODE_POD=hdfs-namenode-default-0
 
-    kubectl exec -n $NAMESPACE $NAME_NODE_POD -- /bin/bash -c "unset HADOOP_OPTS && bin/hdfs dfs -test -e $1"
+    kubectl exec -n $NAMESPACE $NAME_NODE_POD -- /bin/bash -c "unset HADOOP_OPTS && bin/hdfs dfs -test -e /tmp/$1"
```
Comment (maltesander) [inline kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh:20]: You can remove the `unset`. https://github.com/stackabletech/hdfs-operator/pull/148 is merged and should be available in the dev helm repo (it is not tagged and in the stable repo yet)

File: kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+kubectl -n ${NAMESPACE} cp ${JAR_FILE} $SPARK_SLAVE_POD:/tmp
+
+#copy start script to master
+SUBMIT_SCRIPT="${PROJECT_DIR}/scripts/sparkHdfs-submit.sh"
+kubectl -n ${NAMESPACE} cp ${SUBMIT_SCRIPT} $SPARK_MASTER_POD:/tmp
+
+#copy resource file to master
+#RESOURCE_FILE="${PROJECT_DIR}/minimalSpark/src/main/resources/minimalSpark.txt"
+#kubectl -n ${NAMESPACE} cp ${RESOURCE_FILE} $SPARK_MASTER_POD:/tmp
+#kubectl -n ${NAMESPACE} cp ${RESOURCE_FILE} $SPARK_SLAVE_POD:/tmp
```
Comment (maltesander) [inline kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh:29]: remove?

File: kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+#make submit executable
+kubectl exec -n ${NAMESPACE} --stdin $SPARK_MASTER_POD -- /bin/bash -c "chmod 700 /tmp/sparkHdfs-submit.sh"
+
+#temp command until https://github.com/stackabletech/hdfs-operator/pull/148 is deployed
+kubectl exec -n ${NAMESPACE} hdfs-namenode-default-0 -- /bin/bash -x -v -c "unset HADOOP_OPTS"
```
Comment (maltesander) [inline kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh:35]: It is merged and available in helm/dev

File: kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh [hunk 6]
```diff
@@ -11,17 +11,20 @@ function spark_hdfs_file_exist {
...
     if [ $? -eq 0 ]; then
-      #echo "$SPARK_FILE exists."
-        logger::info "Found $EXPECTED_RESULT in $NAME_NODE_POD"
+      #print success message
+        logger::info "The results exist $?. Therefore, found $1 in $NAME_NODE_POD"
     else
        #Print failure message
-        logger::fail  "$EXPECTED_RESULT does not exist on $NAME_NODE_POD."
+        logger::fail  "The results exist $?. Therefore, $1 does not exist on $NAME_NODE_POD."
     fi
```
Comment (maltesander) [inline kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh:29]: I dont like these comments. It is obvious what happens, you can remove them.
Furthermore i would simplify the logging:

    # Validate results
    if [ $? -eq 0 ]; then
        logger::info "$1 exists on $NAME_NODE_POD"
    else
        logger::fail "$1 missing on $NAME_NODE_POD"
    fi

**Review by maltesander — CHANGES_REQUESTED** [review]
Tested locally and works. Just minor things.
stackable-cluster/kuttl-test/tests/spark/sparkPreparation.sh is empty and can be deleted?

From what i see there are some hardcoded (spark) versions, does it make sense to use a variable for that to use with different versions? Same goes for the custom resources of hdfs zk etc.?

File: kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh (hunk 2 above)
Comment (snocke) [reply kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh:17]: Yes. Haven't looked into a better way just yet. Let me know if you have any spontaneous ideas.

File: kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh (hunk 3 above)
Comment (snocke) [reply kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh]: done

File: kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh (hunk 4 above)
Comment (snocke) [reply kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh]: yes. done

File: kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh (hunk 5 above)
Comment (snocke) [reply kuttl-test/tests/spark/sparkHdfs/sparkPreparation.sh]: nice. removal done.

File: kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh (hunk 6 above)
Comment (snocke) [reply kuttl-test/tests/.kubeassert/spark_hdfs_file_exist.sh]: Ok. Agree. Applied your suggestion

**Review by maltesander — APPROVED** [review]
LGTM.

**Comment by snocke** [comment]
Done in Convert to maven pom#42

**Pull request closed without merging**